tests/console/ncurses_extended.pm

```diff
@@ -0,0 +1,131 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Runs several ncurses tests from ncurses package
+# - Text attributes
+# - Colors
+# - Special characters
+# - menus
+# - panels
+# - forms
+# - background combinations
+# - key capture
+# - window movements
+
+# Maintainer: Ivan Lausuch <ilausuch@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+use version_utils 'is_pre_15';
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub send_key_n_times {
+    my ($key, $count) = @_;
+
+    for my $i (1 .. $count) {
+        send_key $key;
+    }
+}
+
+sub send_continue {
+    my ($count) = @_;
+
+    send_key_n_times('ret', $count);
+}
+
+sub auto_ncurses {
+    my ($option, $screen) = @_;
+
+    type_string "./ncurses\n";
+    type_string $option. "\n";
+    assert_screen $screen;
+    send_key 'ctrl-c';
+}
+
+sub run {
+    select_console 'root-console';
+
+    my ($version, $sp) = split(/-/, get_var("VERSION"));
+
+    if (is_pre_15()) {
```

Comment:
that would also call the next line on Leap which is probably invalid. At best you also run a test against openSUSE Leap then :)

Reply:
Thanks @okurz good point. I solved that using is_sle